### PR TITLE
Enum store cleanup 3

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/enumstore.hpp [hunk 1]
```diff
@@ -65,14 +65,18 @@ EnumStoreT<EntryType>::load_unique_value(const void* src, size_t available, Inde
...
     }
     return sizeof(DataType);
 }
 
 template <typename EntryType>
 EnumStoreT<EntryType>::EnumStoreT(bool has_postings)
-    : _store(make_enum_store_dictionary(*this, has_postings, EntryType::hasFold() ? std::make_unique<FoldedComparatorType>(*this) : std::unique_ptr<datastore::EntryComparator>())),
+    : _store(make_enum_store_dictionary(*this, has_postings,
+                                        (has_string_type() ?
+                                         std::make_unique<FoldedComparatorType>(_store.get_data_store()) :
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/attribute/enumstore.hpp]: This does not compile when using clang, due to _store not being initialized:
`
vespa/searchlib/attribute/enumstore.hpp:78:81: error: field '_store' is uninitialized when used here [-Werror,-Wuninitialized]
`

File: vespalib/src/vespa/vespalib/datastore/unique_store_comparator.h [hunk 2]
```diff
@@ -39,51 +39,61 @@ class UniqueStoreFloatingPointComparatorHelper
...
   
-/*
- * Compare two entries based on entry refs.  Valid entry ref is mapped
- * to an entry in a data store.  Invalid entry ref is mapped to a
- * temporary entry referenced by comparator instance.
+/**
+ * Compare two entries based on entry refs.
+ *
+ * Valid entry ref is mapped to an entry in a data store.
+ * Invalid entry ref is mapped to a temporary entry referenced by comparator instance.
```
Comment (toregge) [inline vespalib/src/vespa/vespalib/datastore/unique_store_comparator.h]: The temporary entry is now a part of the comparator instance, so the comment is outdated.

**Review by toregge — APPROVED** [review]

File: searchlib/src/vespa/searchlib/attribute/enumstore.hpp (hunk 1 above)
Comment (geirst) [reply searchlib/src/vespa/searchlib/attribute/enumstore.hpp]: Fixed

File: vespalib/src/vespa/vespalib/datastore/unique_store_comparator.h (hunk 2 above)
Comment (geirst) [reply vespalib/src/vespa/vespalib/datastore/unique_store_comparator.h]: Fixed

**Pull request merged**